field/multiselect/tests/behat/settings_configuration_03.feature

```diff
@@ -22,6 +22,8 @@ Feature: Validate creation and submit for "multiselect" elements using the princ
       | activity  | name           | intro              | course           |
       | surveypro | Surveypro test | For testing backup | Multiselect item |
     And I am on the "Surveypro test" "surveypro activity" page logged in as teacher1
+    And I select "Layout" from secondary navigation
```

Comment:
I think that it's normal that all the tests are failing for you when you use the `I select "xxxxx" from primary/secondary navigation" with Classic.

If you look to core (both 401_STABLE and 402_STABLE) you will see that we are not using that step at all (basically, only in a "boost" specific scenario we use it):

```
$ ag 'from secondary navigation'
theme/boost/tests/behat/settingstabs.feature
22:    When I select "Users" from secondary navigation

lib/tests/behat/behat_navigation.php
1021:     * @When I select :link from secondary navigation
```

So, unless I'm wrong... in order to provide Classic compatibility, something else needs to be used. It may be that we provide straight navigations to this or that page... or it may be that something else is used... but for sure those steps aren't used.

I'll try to find which is the recommendation in those cases... and why those 2 steps exist if they aren't usable...

Reply:
I really wold have the time to fix it. Maybe the straight navigations could be the right solution. Thanks.

Reply:
I think that there are 2 possible solutions for this:

1. Or you use the "page resolvers" that already have in the module (see the `resolve_page_instance_url` implementation).
2. Or you use the admin block, that is always added and then follow the links from there (Like `And I navigate to "XXXX" in current page administration`).

I would say that, in general, using the page resolvers (1) is better and quicker and it's immune to any change in themes. And I would use the administration block (2) only when we really want to verify that manual navigation is working, not when we are testing other features.

Ciao :-)